siphon/simplewebservice/wyoming.py

```diff
@@ -0,0 +1,149 @@
+# Copyright (c) 2013-2015 University Corporation for Atmospheric Research/Unidata.
+# Distributed under the terms of the MIT License.
+# SPDX-License-Identifier: MIT
+"""Read upper air data from the Wyoming archives."""
+
+from ..http_util import HTTPEndPoint
+from io import BytesIO
+import numpy as np
+import pandas as pd
+from metpy.calc import get_wind_components
+
+
+def get_upper_air_data(time, site_id):
+    r"""Download and parse upper air observations from an online archive.
+
+    Parameters
+    ----------
+    time : datetime
+        The date and time of the desired observation.
+
+    site_id : str
+        The three letter ICAO identifier of the station for which data should be
+        downloaded.
+
+    kwargs
+        Arbitrary keyword arguments to use to initialize source
+
+    Returns
+    -------
+        :class:`pandas.DataFrame` containing the data
+
+    """
+    endpoint = WyomingUpperAir('http://weather.uwyo.edu/')
+    fobj = endpoint.get_data(time, site_id)
+    info = endpoint.parse(fobj)
+
+    direction, spd, spd_units = info['wind']
```

Comment:
I think all of the pandas manipulation should be in `get_data` in the WyomingUpperAir class.